src/main/java/org/opensearch/flowframework/common/WorkflowResources.java

```diff
@@ -54,11 +57,11 @@ public enum WorkflowResources {
     /** Workflow steps for deploying/undeploying a model and associated created resource */
     DEPLOY_MODEL(DeployModelStep.NAME, WorkflowResources.MODEL_ID, UndeployModelStep.NAME),
     /** Workflow steps for creating an ingest-pipeline and associated created resource */
-    CREATE_INGEST_PIPELINE(CreateIngestPipelineStep.NAME, WorkflowResources.PIPELINE_ID, null), // TODO delete step
+    CREATE_INGEST_PIPELINE(CreateIngestPipelineStep.NAME, WorkflowResources.PIPELINE_ID, DeleteIngestPipelineStep.NAME),
     /** Workflow steps for creating an ingest-pipeline and associated created resource */
-    CREATE_SEARCH_PIPELINE(CreateSearchPipelineStep.NAME, WorkflowResources.PIPELINE_ID, null), // TODO delete step
+    CREATE_SEARCH_PIPELINE(CreateSearchPipelineStep.NAME, WorkflowResources.PIPELINE_ID, DeleteSearchPipelineStep.NAME),
     /** Workflow steps for creating an index and associated created resource */
-    CREATE_INDEX(CreateIndexStep.NAME, WorkflowResources.INDEX_NAME, NoOpStep.NAME),
+    CREATE_INDEX(CreateIndexStep.NAME, WorkflowResources.INDEX_NAME, DeleteIndexStep.NAME),
```

Comment:
We removed this step because we don't want to let user delete any customer related data from flow framework. 
@amitgalitz can you also take a look here?

Reply:
yeah I think though we have made the decision that its okay to do this with the additional parameter that gets added. Customers in an FGAC domain will still need permission to delete that index from security plugin.

Reply:
> Customers in an FGAC domain will still need permission 

Does that happen automatically or is there further validation that has to be done?

Reply:
happens automatically, would be good to at least manually check :)

Reply:
> We removed this step because we don't want to let user delete any customer related data from flow framework. 

It wasn't quite that broad a decision.  We didn't want a deletion hidden behind an API that did not make the deletion clear, specifically "deprovision".  The addition of a deletion step actually allows users to use that step in a workflow and delete indices (or even delete all indices if they use `*`.  But it will have the word `delete` in the step and be obvious.  And won't give them permission to do anything they couldn't do anyway directly via the API.

Here's the original comment that prompted this feature:
 - https://github.com/opensearch-project/flow-framework/pull/574#discussion_r1525111295